messaging/src/main/java/org/axonframework/messaging/EmptyApplicationContext.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import org.axonframework.configuration.ComponentNotFoundException;
+
+import java.util.Optional;
+
+/**
+ * An {@link ApplicationContext} implementation that does not provide any components.
+ * It is useful as a placeholder or default context when no components are available.
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+public class EmptyApplicationContext implements ApplicationContext {
+
+    @Nonnull
+    @Override
+    public <C> C component(@Nonnull Class<C> type, @Nullable String name) {
+        throw new ComponentNotFoundException(type, name);
```

Comment:
I am wondering whether we should throw a different exception here. If somebody for some reason wired a custom `UnitOfWorkFactory` without providing the `ConfigurationApplicationContext`, all they get is "this component cannot be found." However, what actually occurred, is that they have a no-op `ApplicationContext`, as it can do anything. Hence, I am leaning towards a `UnsupportedOperationException`, clarifying why a component cannot be retrieved from this version of the `ProcessingContext`.

Users may fall in this trap, but our tests also get this `EmptyApplicationContext`. Hence we may as well fall in this trap.

Reply:
I was also considering that, so with your suggestion... I'm going to change that :)

Reply:
I've introduced more changes:
+ there is no "default" ApplicationContext and it's need to be defined while creating the `SimpleUnitOfWork` 
+ I've removed the `ApplicationContext` from the `UnitOfWorkConfiguration`, because the `ApplicationContext` is for components, which are defined in boot time, so there are no sens to allow overriding it per `UnitOfWork` instance in the customization function. 
+ I kept the `UnitOfWorkConfiguration` with just one variable `Executor`. It's more convenient to be able to override it in `UnitOfWorkFactory#create` function with this object in place.